prow/cmd/peribolos/main.go

```diff
@@ -918,9 +918,15 @@ func configureTeamRepos(client teamRepoClient, githubTeams map[string]github.Tea
 
 	actions := map[string]github.RepoPermissionLevel{}
 	for wantRepo, wantPermission := range want {
-		if havePermission, haveRepo := have[wantRepo]; haveRepo && havePermission == wantPermission {
-			// nothing to do
-			continue
+		if havePermission, haveRepo := have[wantRepo]; haveRepo {
+			if havePermission == github.None {
```

Comment:
I'm not sure I like the skip behaviour here, or if this even does what's expected.

In this, we are *specifying* we want a certain permissions level (`wantPermission`) but in certain cases we can't do it. And none doesn't just mean "unparsable", it also means "no access".

Reply:
Thank you for the review. Since the 'have' items come from [GitHub teams API](https://developer.github.com/v3/teams/#list-team-repos), I thought only repos on which the team has some level of access will be listed, therefore 'none' in the above (should?) only refer to 'unparsable' case, otherwise the repo would not have been listed at all. Does that seem incorrect?

That line is an attempt to avoid updating permission levels according to 'want', if the existing permission, i.e. 'have', cannot be understood (as would be with 'Maintain' and 'Triage' levels).

Regarding the skip behaviour, do you mean it is preferable to just fail updating the team, rather than skipping the updating of its permission level on a repo?

Reply:
I think what we should do is we should not put those teams into the `have` bucket at all, just ignore that they were sent to us from the API with `none` at all

Reply:
I think that's probably the best plan, because if we have a specified access, then we hammer it.. but if it's omitted, we don't do anything.

Reply:
Yeah spoke on Slack and not returning items in `ListRepoTeams()` when they have `none` is the best approach since in the normal  V3 API that is not possible anyway